lib/utils/cli.go

```diff
@@ -251,13 +251,21 @@ func InitCLIParser(appName, appHelp string) (app *kingpin.Application) {
 // string that is safe to print on the CLI. This is to ensure that malicious
 // servers can not hide output. For more details, see:
 //   * https://sintonen.fi/advisories/scp-client-multiple-vulnerabilities.txt
-func EscapeControl(s string) string {
-	if needsQuoting(s) {
+func EscapeControl(s string, exceptions ...rune) string {
+	if needsQuoting(s, exceptions...) {
 		return fmt.Sprintf("%q", s)
 	}
 	return s
 }
 
+// AllowNewlines escapes all ANSI escape sequences except newlines from string and returns a
+// string that is safe to print on the CLI. This is to ensure that malicious
+// servers can not hide output. For more details, see:
+//   * https://sintonen.fi/advisories/scp-client-multiple-vulnerabilities.txt
+func AllowNewlines(s string) string {
+	return EscapeControl(s, '\n')
+}
```

Comment:
this will still escape newlines if there are any other control characters present.
something funky like this would work though:
```go
lines := strings.Split(s, "\n")
for i, l := range lines {
  lines[i] = EscapeControl(l)
}
return strings.Join(lines, "\n")
```

Reply:
that's fine, I'm OK escaping the entire string if newlines are present